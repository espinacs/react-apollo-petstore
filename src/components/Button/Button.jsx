import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Button = styled.button`
  /* Adapt the colors based on primary prop */
  background: ${props => props.primary ? "#1689d1" : "white"};
  color: ${props => props.primary ? "white" : "grey"};
  boder-color: ${props => props.primary ? "white" : "#1689d1"};
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid;
  border-radius: 5px;
`;

const StyledButton = ({ children, onClick, primary }) => (
    <Button primary={primary} onClick={onClick}>{children}</Button>
)

StyledButton.propTypes = {
    children: PropTypes.node,
    onClick: PropTypes.func.isRequired,
    primary: PropTypes.bool
}

export default StyledButton