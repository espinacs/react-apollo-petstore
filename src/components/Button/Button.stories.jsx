import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import centered from '@storybook/addon-centered';
import Button from './Button';

import jestTestResults from '../../../.jest-test-results.json';
import { withTests } from '@storybook/addon-jest';

export const wTests = withTests({
  results: jestTestResults,
  filesExt: ""
});

storiesOf('Components|Button', module)
  .addDecorator(wTests('Button'))
  .addDecorator(centered)
  .addDecorator(withKnobs)
  .add('Simple', () => <Button >Click Me</Button>)
  .add('Primary', () => <Button primary>Click Me</Button>)
