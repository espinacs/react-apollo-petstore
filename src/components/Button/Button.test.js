import React from 'react';
import { render, cleanup, Simulate } from 'react-testing-library';
import Button from './';

afterEach(cleanup);

describe('Button', () => {
  const content = 'Click me'
  const onClick = jest.fn()
  test('renders default', () => {
    const props = { onClick };
    const { container, getByText } = render(<Button {...props}>{content}</Button>);
    expect(getByText(content)).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot();
  });
  
  test('renders being primary', () => {
    const props = { onClick, primary: true };
    const { container, getByText } = render(<Button {...props}>{content}</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });
  
  test('is Clicked', () => {
    const props = { onClick };
    const {
      container, getByLabelText, getByText, getByTestId
    } = render(<Button {...props}>{content}</Button>);
    getByText(content).click();
    expect(onClick).toBeCalledTimes(1)

  });
});