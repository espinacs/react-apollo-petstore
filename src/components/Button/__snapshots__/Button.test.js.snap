// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders being primary 1`] = `
<button
  class="sc-bdVaJa fCWzxy"
>
  Click me
</button>
`;

exports[`Button renders default 1`] = `
<button
  class="sc-bdVaJa bgIJis"
>
  Click me
</button>
`;
